core/src/smartcontracts/isi/domain.rs

```diff
@@ -102,6 +102,11 @@ pub mod isi {
                         .expect("should succeed")
                 });
 
+            state_transaction
+                .world
+                .account_permissions
+                .remove(account_id.clone());
```

Comment:
I think this is a good catch, and why not the same for `world.account_roles` ?

Reply:
good point